backend/src/database/dto.rs

```diff
@@ -0,0 +1,80 @@
+use chrono::NaiveDateTime;
+use serde::{Deserialize, Serialize};
+use sqlx::FromRow;
+use uuid::Uuid;
+
+pub type Hash = String;
+
+#[derive(Serialize, Deserialize)]
+pub struct Claim {
+    #[serde(rename = "cTypeHash")]
+    pub ctype_hash: String,
+    contents: serde_json::Value,
+    pub owner: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Credential {
+    pub claim: Claim,
+    #[serde(rename = "claimNonceMap")]
+    claim_nonce_map: serde_json::Value,
+    #[serde(rename = "claimHashes")]
+    claim_hashes: Vec<Hash>,
+    #[serde(rename = "delegationId")]
+    delegation_id: Option<String>,
+    legitimations: Vec<Credential>,
+    #[serde(rename = "rootHash")]
+    pub root_hash: Hash,
+}
+
+#[derive(Serialize, Deserialize, FromRow)]
+pub struct AttestationResponse {
+    pub id: Uuid,
+    pub approved: bool,
+    pub revoked: bool,
+    pub created_at: NaiveDateTime,
+    pub deleted_at: Option<NaiveDateTime>,
+    pub updated_at: Option<NaiveDateTime>,
+    pub ctype_hash: String,
+    pub credential: serde_json::Value,
+    pub claimer: String,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct AttestationRequest {
+    pub ctype_hash: String,
+    pub credential: serde_json::Value,
+    pub claimer: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct UpdateAttestation {
+    pub credential: serde_json::Value,
```

Comment:
not sure if this contains a credential or a claim

Reply:
It is a claim. While the claim is not approved, the user has the possibility to update it. Renamed in: [09b49f9](https://github.com/KILTprotocol/attester/pull/1/commits/09b49f937da86833a0b4f5187078d81b971c946b)